lib/Basic/SourceLoc.cpp

```diff
@@ -836,11 +836,11 @@ static bool isBeforeInSource(
   SourceLoc firstLocInLCA = firstMismatch == firstAncestors.end()
       ? firstLoc
       : sourceMgr.getGeneratedSourceInfo(*firstMismatch)
-          ->originalSourceRange.getEnd();
+          ->originalSourceRange.getStart();
   SourceLoc secondLocInLCA = secondMismatch == secondAncestors.end()
       ? secondLoc
       : sourceMgr.getGeneratedSourceInfo(*secondMismatch)
-          ->originalSourceRange.getEnd();
+          ->originalSourceRange.getStart();
```

Comment:
@hborla This effectively reverts 8c9eeb2edd78efb0e6560f3799cc6e857c66a4b4, AFAIK that shouldn't be necessary with 60da121c1f9fd754cd23eab7d9d4295f8138fe35, let me know if there's a case I'm missing though